docs/getting-started/openshift/index.md

```diff
@@ -0,0 +1,180 @@
+The principal's resource proxy needs to be reachable by your Argo CD API server (argocd-server), which usually runs in the same cluster as the principal. Typically, this service should not be reachable outside the cluster and is exposed using a Kubernetes Service.
+
+```
+argocd-agentctl pki issue resource-proxy \
+    --principal-context <control plane context> \
+    --ip <ip addresses of resource proxy> \
+    --dns <dns names of principal>
+```
+
+Create argocd-redis secret, because principal looks for it to fetch redis authentication details.
+
+```
+oc create secret generic argocd-redis -n argocd --from-literal=auth="$(oc get secret    argocd-redis-initial-password -n argocd -o jsonpath='{.data.admin\.password}' | base64 -d)"
+```
+
+Deploy agent on hub cluster using argocd-operator/gitops-operator using ArgoCD CR given below
+
+
+```
+apiVersion: argoproj.io/v1beta1
+kind: ArgoCD
+metadata:
+  name: argocd
+  Namespace: argocd
+spec:
+  controller:
+    enabled: false
+  argoCDAgent:
+    principal:
+      enabled: true
+      allowedNamespaces: 
+        - "*"
+      jwtAllowGenerate: true
+      auth: "mtls:CN=([^,]+)"
+      logLevel: "trace"
+      image: "quay.io/user/argocd-agent:v1"
+  sourceNamespaces:
+    - "agent-managed"
+    - "agent-autonomous"  					
+```
+
+The above CR should create all the necessary resource for argocd as well as argocd-agent principal in argocd namespace.
+
+
+## Setting up agent workload cluster 
+
+### Configure Argo-CD for Agent 
+ 
+argocd instance on Agent cluster
+
+Creating argocd instance for Workload/spoke cluster, this instance will be using the shared repo, redis instance from principal and, have locally running argocd-application-controller instance.
+Update the below manifest with loadbalancer ip to create argocd instance. 
+```
+  apiVersion: argoproj.io/v1beta1
+  kind: ArgoCD
+  metadata:
+    name: argocd
+    namespace: argocd
+  spec:
+    repo:
+      remote: <repo LoadBalancer IP addr>:8081
+    redis:
+      remote: <redisLoadBalancer IP addr>:6379
+    server:
+      enabled: false
+    sourceNamespaces:
+    - <workload namespace>
+```
+
+### Configure Agent in managed mode
+
+Before installing agent resources create 
+Create a TLS secret containing the issued certificate for agent
+
+Create the PKI on the agent:
+Run this command while connected to principal
+```
+argocd-agentctl pki issue agent <agent-name> --agent-context <workload context> --agent-namespace <workload namespace> --upsert
+```
+
+Apply the installation manifests for argocd agent
+```
+oc apply -n $(workload-namespace) -k 'https://github.com/argoproj-labs/argocd-agent/install/kubernetes/agent?ref=main'
+```
+This should create all the required agent related resources.
+
+Update the configMap with name `argocd-agent-params`  with parameters related to agent.mode,agent.creds, agent.namespace, agent.server.address.	
+```
+  agent.mode: autonomous
+  agent.creds: mtls:any
```

Comment:
A little different from what I used while testing:

```
data:
  agent.keep-alive-ping-interval: 50s # Otherwise the connection would close I guess
  agent.tls.client.insecure: 'true' # This is for testing purpose
  agent.server.port: '443'
  agent.tls.root-ca-path: ''
  agent.tls.client.cert-path: ''
  agent.tls.root-ca-secret-name: argocd-agent-ca
  agent.tls.secret-name: argocd-agent-client-tls
  agent.mode: autonomous
  agent.log.level: info
  agent.namespace: argocd
  agent.server.address: <principal server address>
  agent.metrics.port: '8181'
  agent.tls.client.key-path: ''
  agent.healthz.port: '8002'
  agent.creds: 'mtls:any'
```

Reply:
```
  agent.healthz.port: '8002'    <------ this was missing
```